secrets/secrets-kubernetes/main.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
+	"github.com/wasmCloud/wasmCloud-contrib/secrets/secrets-kubernetes/pkg/secrets"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	ServiceName = "kube"
+)
+
+type kubeSecretsServer struct {
+	kubeClient client.Client
+}
+
+func (s *kubeSecretsServer) Locator(name string) (string, string, error) {
+	parts := strings.Split(name, "/")
+	if len(parts) != 2 {
+		return "", "", secrets.ErrInvalidRequest
+	}
+
+	return parts[0], parts[1], nil
+}
+
+func (s *kubeSecretsServer) Get(ctx context.Context, r *secrets.Request) (*secrets.Response, error) {
+	fmt.Printf(">>> GET\n%s\n", r.String())
+	kubeSecretName, kubeEntryName, err := s.Locator(r.Name)
+	if err != nil {
+		return r.Reject(fmt.Errorf("%w: invalid secret name format", secrets.ErrInvalidRequest)), nil
+	}
+	slog.Info("secret name", slog.Any("name", kubeSecretName), slog.Any("key", kubeEntryName))
+
+	kubeSecret := corev1.Secret{}
+	if err = s.kubeClient.Get(context.Background(), client.ObjectKey{
+		Namespace: "default",
+		Name:      kubeSecretName,
+	}, &kubeSecret); err != nil {
+		slog.Info("kubee rr", slog.Any("err", err))
+		return r.Reject(fmt.Errorf("%w: %s", secrets.ErrInvalidRequest, err)), nil
+	}
+
+	kubeEntryValue, ok := kubeSecret.Data[kubeEntryName]
+	if !ok {
+		slog.Info("kubee key err")
+		return r.Reject(fmt.Errorf("%w: secret not found", secrets.ErrInvalidRequest)), nil
+	}
+
+	return r.Reply(&secrets.SecretValue{
+		Name:         r.Name,
+		StringSecret: string(kubeEntryValue),
+		Version:      kubeSecret.ResourceVersion,
+	}), nil
+}
+
+func main() {
+	mainCtx, mainCancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM, syscall.SIGUSR1, syscall.SIGUSR2, syscall.SIGHUP)
+	defer mainCancel()
+
+	kubeClient, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		slog.Error("Couldn't setup kube client", slog.Any("error", err))
+		os.Exit(1)
+	}
+
+	s := &kubeSecretsServer{
+		kubeClient: kubeClient,
+	}
+
+	nc, err := nats.Connect(nats.DefaultURL)
+	if err != nil {
+		slog.Error("Couldn't setup nats client", slog.Any("error", err))
+		os.Exit(1)
+	}
+
+	errorCallback := func(_ *nats.Msg, err error) {
+		slog.Error("server error", slog.Any("error", err))
+	}
+
+	secretBackendKey, err := nkeys.FromSeed([]byte("SXABDJLREH6OP7LQAYYPONTUUZRD4SO5QPJOE4X5FC57ZTUE2PDFRA5FSM"))
+	if err != nil {
+		slog.Error("Couldn't setup xkey", slog.Any("error", err))
+		os.Exit(1)
+	}
+
+	secretsServer, err := secrets.NewServer(ServiceName,
+		nc,
+		s,
```

Comment:
Would it make sense for the sake of consistency to wrap these in ServerOption as well so that you could potentially pass mocks/stubs in as part of testing?